bika/lims/browser/fields/aranalysesfield.py

```diff
@@ -116,6 +118,10 @@ def set(self, instance, service_uids, prices=None, specs=None, **kwargs):
 
         assert type(service_uids) in (list, tuple)
 
+        for i, item in enumerate(service_uids):
+            if hasattr(item, 'getServiceUID'):
+                service_uids[i] = item.getServiceUID()
```

Comment:
It would be better to check explicitly for the type rather than if an attribute exists, and support brains and AnalysisService objects as well:
```python
obj = api.get_object(item)
if IAnalysisService.providedBy(obj):
    service_uids[i] = api.get_uid(obj)
elif IAnalysis.providedBy(obj) and IRequestAnalysis.providedBy(obj):
    service_uids[i] = obj.getServiceUID()
```

Reply:
I didn't add 'IRequestAnalysis' and I wonder if we really need it. Since Analyses field, isn't it enough if it accepts Analyses and AS objects?

Reply:
Yes, we need to check for `IRequestAnalysis` because there are some types of Analyses (`ReferenceAnalysis` type) that don't belong to ARs, rather to `SampleReference`